x-pack/plugin/ingest/src/main/java/org/elasticsearch/xpack/ingest/UrlPartsProcessor.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.ingest;
+
+import org.elasticsearch.ingest.AbstractProcessor;
+import org.elasticsearch.ingest.ConfigurationUtils;
+import org.elasticsearch.ingest.IngestDocument;
+import org.elasticsearch.ingest.Processor;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+
+public class UrlPartsProcessor extends AbstractProcessor {
+
+    public static final String TYPE = "url_parts";
+
+    private final String field;
+    private final String targetField;
+    private final boolean removeIfSuccessful;
+    private final boolean keepOriginal;
+
+    UrlPartsProcessor(String tag, String description, String field, String targetField, boolean removeIfSuccessful, boolean keepOriginal) {
+        super(tag, description);
+        this.field = field;
+        this.targetField = targetField;
+        this.removeIfSuccessful = removeIfSuccessful;
+        this.keepOriginal = keepOriginal;
+    }
+
+    public String getField() {
+        return field;
+    }
+
+    public String getTargetField() {
+        return targetField;
+    }
+
+    public boolean getRemoveIfSuccessful() {
+        return removeIfSuccessful;
+    }
+
+    public boolean getKeepOriginal() {
+        return keepOriginal;
+    }
+
+    @Override
+    public IngestDocument execute(IngestDocument ingestDocument) throws Exception {
+        String value = ingestDocument.getFieldValue(field, String.class);
+
+        URL url;
+        try {
+            url = new URL(value);
+        } catch (MalformedURLException e) {
+            throw new IllegalArgumentException("unable to parse URL [" + value + "]");
+        }
+        var urlParts = new HashMap<String, Object>();
+        urlParts.put("domain", url.getHost());
```

Comment:
ECS isn't clear on what's correct here, but what does `getHost` return for bracked IPv6 addresses?

@elastic/ecs Should `url.domain` include the brackets that are required when using IPv6 addresses in URLs? https://www.ietf.org/rfc/rfc2732.txt

Reply:
Per [this source](https://docs.oracle.com/en/java/javase/14/docs/api/java.base/java/net/URL.html#getHost()), `getHost()` will return the IPv6 address enclosed in the brackets.

Since brackets are required with a literal IPv6 address, `url.domain` should include the brackets. We can improve the description of `url.domain` in the ECS docs to clarify.